src/examples/libpmemobj/cpp_map/ctree_map_persistent.hpp

```diff
@@ -203,15 +203,15 @@ class ctree_map_p {
 	{
 		auto pop = nvobj::pool_by_vptr(this);
 		nvobj::transaction::exec_tx(pop, [&] {
-			if (root->inode) {
-				root->inode->clear();
-				nvobj::delete_persistent<node>(root->inode);
-				root->inode = nullptr;
+			if (this->root->inode) {
```

Comment:
the [&] lambda capture means that you can you 'this' inside of it. I'm certain that you don't need the explicit 'this->'. And if you do, please consult that with Tomek and change this everywhere in the codebase.